dipy/align/mattes.pyx

```diff
@@ -0,0 +1,1276 @@
+#!python
+#cython: boundscheck=False
+#cython: wraparound=False
+#cython: cdivision=True
+
+import numpy as np
+cimport numpy as cnp
+cimport cython
+from fused_types cimport floating
+
+from dipy.align.vector_fields cimport(_apply_affine_3d_x0,
+                                      _apply_affine_3d_x1,
+                                      _apply_affine_3d_x2,
+                                      _apply_affine_2d_x0,
+                                      _apply_affine_2d_x1)
+
+from dipy.align.transforms cimport (Transform)
+
+cdef extern from "dpy_math.h" nogil:
+    double cos(double)
+    double sin(double)
+    double log(double)
+
+class MattesBase(object):
+    def __init__(self, nbins):
+        r""" MattesBase
+        Base class for the Mattes' Mutual Information metric.
```

Comment:
Also: `Mattes's`, if this is a possessive, or `Mattes` (no apostrophe) if not. I think the latter.